crates/bevy_render/src/camera/camera.rs

```diff
@@ -36,14 +36,51 @@ use wgpu::Extent3d;
 pub struct Viewport {
     /// The physical position to render this viewport to within the [`RenderTarget`] of this [`Camera`].
     /// (0,0) corresponds to the top-left corner
-    pub physical_position: UVec2,
+    pub physical_position: AbsOrPercVec,
     /// The physical size of the viewport rectangle to render to within the [`RenderTarget`] of this [`Camera`].
     /// The origin of the rectangle is in the top-left corner.
-    pub physical_size: UVec2,
+    pub physical_size: AbsOrPercVec,
     /// The minimum and maximum depth to render (on a scale from 0.0 to 1.0).
     pub depth: Range<f32>,
 }
 
+/// A vector wrapper that represents either absolute value or value as a percentage.
+///
+/// For example, used in representing viewport position, either as absolute pixel value, or as a
+/// percentage of render target size.
+#[derive(Reflect, Debug, Clone, Serialize, Deserialize)]
+pub enum AbsOrPercVec {
```

Comment:
```suggestion
pub enum ViewportSize {
```

I find the current name very hard to read. I'd much prefer to either use something tightly scoped like this. Down the line we can consider integrating this into a more generic `Size` and sharing it with `bevy_ui`.

Reply:
I considered `ViewportSize`, but it's also used as position so I wasn't too convinced (I thought it would be confusing). 
So I used a temporary, obnoxiously bad name to get better ideas from reviewers.

Reply:
IMO we should have two distinct types then: `ViewportSize` and `ViewportPosition`. This makes it much easier to understand what's going on in the code, and prevents us from using them in the wrong place by accident.

Reply:
(This is part of why I dislike using raw `UVec2` in the first place)

Reply:
But then we have two types that are exactly the same, which isn't perfect either (even if only for code quality).

Reply:
The types *mean* different things though. Perhaps they could both wrap the same internal data?

Reply:
I'd argue that type is just the data and functions it provides (and those are the same between size and position) and context is what gives them meaning.
But `winit` does have separate types for position and size, so I guess we can too :p 
Sharing same internal data may be a nice solution

Reply:
But then how to name this internal data? :thinking: